src/kernel.cpp

```diff
@@ -373,7 +373,15 @@ bool ComputeNextStakeModifier(const CBlockIndex* pindexPrev, uint64_t& nStakeMod
 // Get stake modifier checksum
 unsigned int GetStakeModifierChecksum(const CBlockIndex* pindex)
 {
-    assert (pindex->pprev || pindex->GetBlockHash() == (!fTestNet ? hashGenesisBlock : hashGenesisBlockTestNet));
+    if (pindex->pprev == nullptr && pindex != pindexGenesisBlock) 
+    {
```

Comment:
This is leaving out the hash check. Checking the pointer is correct is not the same as checking the hash of the genesis block index pointed to by the pointer.